### PR TITLE
NyxLex - Verify add button functionality

### PR DESCRIPTION
**Pull request by NyxLex**

### CODE REVIEW

**Review by Percifalll — APPROVED** [review]

File: src/test/java/com/ss/ita/rozetka/BasketTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.ss.ita.rozetka.ui.pages.ProductTypePage;
+import org.testng.annotations.Test;
+
+import static com.ss.ita.rozetka.ui.util.PageUtil.getCurrentUrl;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class BasketTest {
+    @Test
+    public void verifyAddButtonFunctionality() {
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/BasketTest.java:17]: add product

File: src/test/java/com/ss/ita/rozetka/BasketTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                .open()
+                .openProductCategoryPage(GeneralProductCategory.NOTEBOOKS_AND_COMPUTERS)
+                .openProductTypePage(ProductCategoryAndSubCategory.NOTEBOOKS);
+
+        assertThat(getCurrentUrl())
+                .as("Notebooks subcategory page should be opened")
+                .contains("https://rozetka.com.ua/notebooks/c80004/");
+
+        BasketModal basketModal = productTypePage
+                .openProductPage(0)
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/BasketTest.java:28]: either openProduct(0) or openProductPageForProduct(0)

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/ss/ita/rozetka/BasketTest.java (hunk 2 above)
Comment (toleksyn) [reply src/test/java/com/ss/ita/rozetka/BasketTest.java:28]: re-do method to start with 1

File: src/main/java/com/ss/ita/rozetka/ui/pages/HeaderPage.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+import com.ss.ita.rozetka.pageobject.elements.Header;
+
+public abstract class HeaderPage {
+    protected Header header = new Header();
+
+    public Header getHeader(){
+        return header;
+    }
+}
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/HeaderPage.java:11]: this shouldn't be here, sync your branch with main properly

**Review by toleksyn — CHANGES_REQUESTED** [review]

**Review by Percifalll — APPROVED** [review]

**Review by anastasia8755 — APPROVED** [review]

File: src/main/java/com/ss/ita/rozetka/pageobject/modals/BasketModal.java [hunk 4]
```diff
@@ -90,4 +93,25 @@ public T close() {
         $x("//button[contains(@class, 'modal__close')]").click();
         return pageObject;
     }
+
+    @Step("BasketModal: increase amount on {numberOfProducts} product(s)")
+    public BasketModal increaseAmountOfProduct(int numberOfProducts, int sequenceNumberOfProduct) {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/modals/BasketModal.java:98]: productNumber, increaseNumber

File: src/main/java/com/ss/ita/rozetka/pageobject/modals/BasketModal.java [hunk 5]
```diff
@@ -90,4 +93,25 @@ public T close() {
...
+        SelenideElement increaseButton = $x(format("(//button[contains(@class,'cart-counter__button')])[%d]", specificNumber));
+        for (int i = 0; i < numberOfProducts; i++) {
+            increaseButton.click();
+            waitForTotalPriceToUpdate(getProductsTotalPrice());
+        }
+        return this;
+    }
+
+    @Step("BasketModal: decrease amount on {number} product(s)")
+    public BasketModal decreaseAmountOfProduct(int numberOfProducts, int sequenceNumberOfProduct) {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/modals/BasketModal.java:109]: productNumber, decreaseNumber

**Review by toleksyn — CHANGES_REQUESTED** [review]

**Pull request merged**